sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -288,31 +288,36 @@ private Mono<ConfidentialClientApplication> getManagedIdentityConfidentialClient
                 + "/" + tenantId;
 
             // Temporarily pass in Dummy Client secret and Client ID. until MSal removes its requirements.
-            IClientCredential credential = ClientCredentialFactory.createFromSecret("dummy-secret");
+            IClientCredential credential = ClientCredentialFactory
+                .createFromSecret(clientSecret != null ? clientSecret : "dummy-secret");
             ConfidentialClientApplication.Builder applicationBuilder =
-                ConfidentialClientApplication.builder(clientId == null ? IdentityConstants.DEVELOPER_SINGLE_SIGN_ON_ID
+                ConfidentialClientApplication.builder(clientId == null ? "SYSTEM-ASSIGNED-MANAGED-IDENTITY"
```

Comment:
should this be added to `IdentityConstants`?

Reply:
This is not a valid client id, and gets used only once, we can keep it local.
IdentityConstants holds valid variables currently.